### DISCUSSION TITLE
Composing tasks with `series` function

### DISCUSSION THREAD

**Discussion by keithbentrup** [gulpjs/gulp#2694] [Help] [answered]
Before you open this issue, please complete the following tasks:

* [x] use the search bar at the top of the page to search this repository for similar issues or discussions that have already been opened.
* [x] if you are looking for help from the gulp team or community, open a [discussion](https://github.com/gulpjs/gulp/discussions).
* [x] if you think there is a problem with the plugin you're using, open a [discussion](https://github.com/gulpjs/gulp/discussions).
* [x] if you think there is a bug in our code, open this issue.

### What were you expecting to happen?
The series task to complete successfully.

### What actually happened?
Fails with error in screenshot: "Did you forget to signal async completion?"

### Please give us a sample of your gulpfile

```js
import gulp from 'gulp'

export const someTaskA = function someTaskA(cb) {
  cb()
}

export const someTaskB = () => Promise.resolve()

export const someTaskC = () => gulp.src('test-file').pipe(gulp.dest('tmp'))

export const test = () => gulp.series(someTaskA)

```

### Terminal output / screenshots

![image](https://user-images.githubusercontent.com/253579/188483842-5b511d42-77f8-4344-a034-8b74b0c57399.png)

### Please provide the following information:
* OS & version [e.g. MacOS Catalina 10.15.4]: 12.3
* node version (run `node -v`): 14.x, 16.x, 18.x (using fnm)
* npm version (run `npm -v`): 8.15.0
* gulp version (run `gulp -v`): 4.0.2

### Additional information
I see many people have had similar error messages, so I tried to break it down into the simplest case(s). I'm using ESM style whereas most examples I see are in CJS. Do I really have something wrong with these tasks? They work fine individually, but just not when invoked individually (any of them) in a series. I switched over to ESM from CJS style because one of the gulp plugins (that I've been using) requires it now. I wondering if I'm running into a case similar to the "[forward references](https://gulpjs.com/docs/en/api/series/#forward-references)" and if so, what is the right syntax here because I've tried so many variations without success.

**Comment by phated** [comment, marked as answer]
`export const test = () => gulp.series(someTaskA)` is incorrect. Gulp's `series` and `parallel` are composition functions and you should not be wrapping them in another function. e.g. `export const test = gulp.series(someTaskA)`

**Reply by keithbentrup** [reply]
Doh! I had defined all my other tasks with functions that I didn't notice that series tasks are not defined that way. Thanks!

**Reply by phated** [reply]
Fwiw, it's **never** recommended to use arrow functions to define tasks. Named functions must be used for task names to display correctly when composing.
